feat(buildDockerAndPublishImage): allow to specify a target when using docker bake

lemeurherve opened the pull request:
This PR adds a `dockerBakeTarget` parameter to the `buildDockerAndPublishImage` function to allow specifying a target when using docker bake.

This PR also fixes the fact that before this change when using docker bake, cst tests are executed only on the image with the `latest` tag instead of testing all<sup>*</sup> built targets.

Ref:
- https://github.com/jenkins-infra/helpdesk/issues/3887

*: with a platform corresponding to the architecture of the machine executing the tests.

lemeurherve commented:
Tested with success in https://github.com/jenkins-infra/docker-jenkins-weekly/pull/1467:

https://infra.ci.jenkins.io/job/docker-jobs/job/docker-jenkins-weekly/job/PR-1467/3/

dduportal commented on vars/buildDockerAndPublishImage.txt:
```diff
@@ -18,6 +18,7 @@ The following arguments are available for this function:
   * String **registry**: (Optional, defaults to "" which defines the registry based on the current controller setup) Container registry to deploy this image to (Example: "ghcr.io", "python/2.7").
   * String **unstash**: (Optional, default to "") Allow to restore files from a previously saved stash if not empty, should contain the name of the last stashed as per https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#unstash-restore-files-previously-stashed.
   * String **dockerBakeFile**: (Optionan, default to "") Specify the path to a custom Docker Bake file to use instead of the default one
+  * String **dockerBakeTarget**: (Optionan, default to "default") Specify the target of a custom Docker Bake file to work with
```
```suggestion
  * String **dockerBakeTarget**: (Optional, default to "default") Specify the target of a custom Docker Bake file to work with
```

(nit, looks like copy and paste of a pre-existing type)

dduportal reviewed (dismissed):
(edited after team discussion)

> This PR also fixes the fact that before this change when using docker bake, cst tests are executed only on the image with the latest tag instead of testing all* built targets.

~This is a distinct change which need to be discussed as there were reason for not doing it.~

~Can you remove it from this PR please as it is a blocker from my point of view.~

---- 

After discussion:

- I misunderstood: the changes on "tests" in this PR not only are required, but are also **only**  scoped to choose the proper local image to build. The PR body explains it carefully, I misread the asterisk, nevermind my remark
- Although we want to remove the naming constraint "image tag" <-> "target" (ref. https://github.com/jenkins-infra/docker-jenkins-weekly/pull/1467)

lemeurherve commented:
> * Although we want to remove the naming constraint "image tag" <-> "target" (ref. [Test jenkins-infra/pipeline-library#822 docker-jenkins-weekly#1467](https://github.com/jenkins-infra/docker-jenkins-weekly/pull/1467))

Addressed in https://github.com/jenkins-infra/docker-jenkins-weekly/commit/814c2a0c84a4331b04132f98b6e13842a8062994 by retrieving the tag of the selected target(s), with the `/` character replaced by `#` so it can be used as Makefile step name.

Tested successfully in https://github.com/jenkins-infra/docker-jenkins-weekly/pull/1467 from where I removed the tags modifications to keep only the test of this PR:
- With a specific target: https://infra.ci.jenkins.io/job/docker-jobs/job/docker-jenkins-weekly/job/PR-1467/13/console
- With default target (no dockerBakeTarget specified): https://infra.ci.jenkins.io/job/docker-jobs/job/docker-jenkins-weekly/job/PR-1467/14/console

dduportal reviewed (approved):
LGTM!
